authorization-service/src/functions/basicAuthorizer/handler.ts

```diff
@@ -0,0 +1,35 @@
+import 'source-map-support/register';
+import { APIGatewayAuthorizerCallback, APIGatewayTokenAuthorizerEvent } from 'aws-lambda';
+import { ALLOW, DENY, generatePolicy, TOKEN, UNAUTHORIZED } from '@functions/basicAuthorizer';
+
+export const basicAuthorizer = async (
```

Comment:
Please note that in TS you can use generics for a better experience with the types
```
import {
  APIGatewayAuthorizerResult,
  APIGatewayTokenAuthorizerEvent,
  Handler,
} from 'aws-lambda';
export const basicAuthorizer: Handler<
  APIGatewayTokenAuthorizerEvent,
  APIGatewayAuthorizerResult
> = ...
```

Reply:
It seems that `async` is redundant since the handler operates with callbacks.

Reply:
In order not to add type to each parameter you can add generic:
```
import { APIGatewayTokenAuthorizerHandler } from 'aws-lambda';
export const basicAuthorized: APIGatewayTokenAuthorizerHandler = () => {...
```